pkg/mimirtool/commands/alerts.go

```diff
@@ -109,9 +111,33 @@ func (a *AlertmanagerCommand) getConfig(_ *kingpin.ParseContext) error {
 		return err
 	}
 
-	p := printer.New(a.DisableColor)
+	if a.OutputDir == "" {
+		p := printer.New(a.DisableColor)
+		return p.PrintAlertmanagerConfig(cfg, templates)
+	}
+	return a.outputAlertManagerConfigTemplates(cfg, templates)
+}
 
-	return p.PrintAlertmanagerConfig(cfg, templates)
+func (a *AlertmanagerCommand) outputAlertManagerConfigTemplates(config string, templates map[string]string) error {
+	var baseDir string
+	baseDir, err := filepath.Abs(a.OutputDir)
+	if err != nil {
+		return err
+	}
+	log.Debug(fmt.Sprintf("writing the config file to %s", filepath.Join(baseDir, "config.yaml")))
+	err = os.WriteFile(filepath.Join(baseDir, "config.yaml"), []byte(config), os.FileMode(0o666))
```

Comment:
Let's default to making the file a bit less permissive by default. I'm thinking about running on shared machines and the config can contain API keys and such. Should we go with `0o600`?
```suggestion
	log.Debug(fmt.Sprintf("writing the config file to %s", filepath.Join(baseDir, "config.yaml")))
	err = os.WriteFile(filepath.Join(baseDir, "config.yaml"), []byte(config), os.FileMode(0o600))
```

Reply:
You are right, not even sure why I was thinking of using `0o666` in the first place. Updated.